src/bin/pg_autoctl/pgctl.c

```diff
@@ -181,6 +181,10 @@ config_find_pg_ctl(PostgresSetup *pgSetup)
 	{
 		char *program = pg_ctls[0];
 		char *version = pg_ctl_version(program);
+		if (version == NULL) {
+			log_fatal("Found %s, but failed to get version", program);
```

Comment:
I think I would rephrase as "Failed to get output from `%s version`". It's a very rare event, but at least showing what we tried might allow the user to copy/paste and see for themselves, either the same problem, or that we have a bug somewhere.